kafka-rest/src/test/java/io/confluent/kafkarest/resources/v3/ProducerMetricsTest.java

```diff
@@ -152,4 +152,23 @@ public void testWindowedCountMetrics() throws Exception {
       assertEquals(10.0, mBeanServer.getAttribute(beanNames.iterator().next(), metric));
     }
   }
+
+  @Test
+  public void testWindowedSumMetrics() throws Exception {
```

Comment:
nit: It would be nice to also have a test (probably in `ProduceActionTest` that sets up messages one of which gets rate-limited and another of which causes an HTTP 400, and assert that the former doesn't bump the metrics while the latter does (as well as all the valid, non-rate-limited messages inbetween). That's entirely optional though.

Reply:
I agree, but I'm going to leave it for now and try and get the main billing flow working 

I've raised the following so we don't forget about it https://confluentinc.atlassian.net/browse/KREST-5786